modules/api/src/main/java/org/apache/ignite/query/sql/async/AsyncSqlMultiResultSet.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.query.sql.async;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Asynchronous multi-statement result set.
+ */
+public interface AsyncSqlMultiResultSet extends AsyncSqlResultSet {
+    /**
+     * Returns current statement result index.
+     *
+     * @return Statement result index.
+     */
+    int currentResult();
+
+    /**
+     * Skip current statement result and fetch first page of the next statement.
+     *
+     * @return Operation future.
+     */
+    CompletionStage<AsyncSqlMultiResultSet> skipResult();
```

Comment:
```suggestion
    CompletionStage<AsyncSqlMultiResultSet> skipResultAsync();
```

Should we add `Async` suffix to all async methods?